source-code/badge/src/badge.ts

```diff
@@ -28,21 +26,20 @@ export const badge = async (url: string) => {
 		return fromCache
 	}
 
-	// initialize a new file system on each request to prevent cross request pollution
-	const fs = createNodeishMemoryFs()
-	await cloneRespository(url, fs)
+	// initialize a lisa repo instance on each request to prevent cross request pollution
+	const repo = await open(url, { nodeishFs: createNodeishMemoryFs() })
 
 	// Set up the environment functions
 	const env: InlangEnvironment = {
```

Comment:
InlangEenvironment shouldn't be needed anymore. But, we can fix that in a follow up commit.